src/services/moderation.service.ts

```diff
@@ -275,6 +274,53 @@ export class ModService {
     }
   }
 
+  /// Bans the user from reading/sending
+  /// in specified channels.
+  /// Files a report about it.
+  public async channelBan(
+    guild: Guild,
+    username: string,
+    channels: GuildChannel[]
+  ): Promise<GuildChannel[]> {
+    const uid = Moderation.Helpers.resolveUser(guild, username);
+
+    if (!uid) {
+      this._loggerService.error(`Failed to resolve ${username} to a user.`);
+      return [];
+    }
+
+    const ret: GuildChannel[] = [];
+    const proms = [];
+    for (const chan of channels) {
+      this._loggerService.debug(`Taking channel permissions away in ${chan.name}`);
+      proms.push(
+        chan
+          .overwritePermissions(uid, { READ_MESSAGES: false, SEND_MESSAGES: false })
+          .then(() => ret.push(chan))
+          .catch((ex) => {
+            this._loggerService.error(
+              `Failed to adjust permissions for ${username} in ${chan.name}`,
+              ex
+            );
+          })
+      );
+    }
+
+    await Promise.all(proms);
+
+    this._insertReport(
```

Comment:
await

Reply:
my intention was to not block on reporting